pkg/driver/dra_hooks.go

```diff
@@ -312,9 +324,69 @@ func (np *NetworkDriver) prepareResourceClaim(ctx context.Context, claim *resour
 				routeCfg.Source = route.Src.String()
 			}
 			routeCfg.Scope = uint8(route.Scope)
+			routeCfg.Table = route.Table
 			podCfg.NetworkInterfaceConfigInPod.Routes = append(podCfg.NetworkInterfaceConfigInPod.Routes, routeCfg)
 		}
 
+		// Obtain the rules associated to the interface
+		rules, err := nlHandle.RuleList(netlink.FAMILY_ALL)
+		if err != nil {
+			klog.Infof("Failed to get ip rules for interface %s : %v", ifName, err)
+		}
+		addrs, err := nlHandle.AddrList(link, netlink.FAMILY_ALL)
+		if err != nil {
+			klog.Infof("Failed to get ip addresses for interface %s : %v", ifName, err)
+		}
+		klog.V(5).Infof("Checking for rules for interface %s", ifName)
+		for _, rule := range rules {
```

Comment:
I don't think there's any other way to do this right? It feels wasteful to have to look through every rule and IP and see if they are associated. Can we create some map from the rules since that part doesn't change between routes

Reply:
Sorry I recognize the previous iteration was not optimized at all. I've made the adjustments.

Reply:
Thanks!